pkg/component/etcd/etcd/crd_test.go

```diff
@@ -45,22 +50,70 @@ var _ = Describe("CRD", func() {
 
 	DescribeTable("CRD is deployed",
 		func(crdName string) {
-			Expect(c.Get(ctx, client.ObjectKey{Name: crdName}, &apiextensionsv1.CustomResourceDefinition{})).To(Succeed())
+			verifyDeployedCRD(ctx, crdName, c)
 		},
 
-		Entry("Etcd", "etcds.druid.gardener.cloud"),
-		Entry("EtcdCopyBackupsTask", "etcdcopybackupstasks.druid.gardener.cloud"),
+		Entry("Etcd", druidcorecrds.ResourceNameEtcd),
+		Entry("EtcdCopyBackupsTask", druidcorecrds.ResourceNameEtcdCopyBackupsTask),
 	)
 
 	DescribeTable("should re-create CRD if it is deleted",
 		func(crdName string) {
 			Expect(c.Delete(ctx, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: crdName}}, &client.DeleteOptions{})).To(Succeed())
 			Expect(c.Get(ctx, client.ObjectKey{Name: crdName}, &apiextensionsv1.CustomResourceDefinition{})).To(BeNotFoundError())
 			Expect(crdDeployer.Deploy(ctx)).To(Succeed())
-			Expect(c.Get(ctx, client.ObjectKey{Name: crdName}, &apiextensionsv1.CustomResourceDefinition{})).To(Succeed())
+			verifyDeployedCRD(ctx, crdName, c)
 		},
 
-		Entry("Etcd", "etcds.druid.gardener.cloud"),
-		Entry("EtcdCopyBackupsTask", "etcdcopybackupstasks.druid.gardener.cloud"),
+		Entry("Etcd", druidcorecrds.ResourceNameEtcd),
+		Entry("EtcdCopyBackupsTask", druidcorecrds.ResourceNameEtcdCopyBackupsTask),
 	)
+
+	Describe("CRD is destroyed", func() {
+		JustBeforeEach(func() {
+			Expect(crdDeployer.Deploy(ctx)).To(Succeed())
+		})
+
+		DescribeTable("CRD is deleted",
+			func(crdName string) {
+				Expect(c.Delete(ctx, &apiextensionsv1.CustomResourceDefinition{ObjectMeta: metav1.ObjectMeta{Name: crdName}}, &client.DeleteOptions{})).To(Succeed())
+				Expect(c.Get(ctx, client.ObjectKey{Name: crdName}, &apiextensionsv1.CustomResourceDefinition{})).To(BeNotFoundError())
+			},
+
+			Entry("Etcd", druidcorecrds.ResourceNameEtcd),
+			Entry("EtcdCopyBackupsTask", druidcorecrds.ResourceNameEtcdCopyBackupsTask),
+		)
+	})
+
+	DescribeTable("Get CRD",
+		func(crdName string) {
+			crd, err := crdDeployer.GetCRD(crdName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(crd).NotTo(BeNil())
+		},
+
+		Entry("Etcd", druidcorecrds.ResourceNameEtcd),
+		Entry("EtcdCopyBackupsTask", druidcorecrds.ResourceNameEtcdCopyBackupsTask),
+	)
+
+	DescribeTable("Get CRD YAML",
+		func(crdName string) {
+			crd, err := crdDeployer.GetCRDYaml(crdName)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(crd).NotTo(BeNil())
+		},
+
+		Entry("Etcd", druidcorecrds.ResourceNameEtcd),
+		Entry("EtcdCopyBackupsTask", druidcorecrds.ResourceNameEtcdCopyBackupsTask),
+	)
+
 })
+
+func verifyDeployedCRD(ctx context.Context, crdName string, c client.Client) {
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	err := c.Get(ctx, client.ObjectKey{Name: crdName}, crd)
+	Expect(err).NotTo(HaveOccurred())
+	deletionProtectedVal, ok := crd.GetLabels()[gardenerutils.DeletionProtected]
+	Expect(ok).To(BeTrue())
+	Expect(deletionProtectedVal).To(Equal("true"))
+}
```

Comment:
Replace this func with 

```golang
	crd := &apiextensionsv1.CustomResourceDefinition{}
	Expect(c.Get(ctx, client.ObjectKey{Name: crdName}, crd)).To(Succeed())
	Expect(crd.Labels).To(HaveKeyWithValue("gardener.cloud/deletion-protected", "true"))
```

inline

Reply:
addressed in https://github.com/gardener/gardener/pull/11545/commits/2a31baa0681d92dc43a18ce1a27a2d9cbe4e45f8

Reply:
Let's keep it as it is. I closed https://github.com/gardener/gardener/pull/11685